frontend/src/pages/journal/components/note/note.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  changeCursorPosition,
+  debounce,
+  getCursorPosition,
+  sanitizeInput,
+} from '#libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFormController,
+  useParams,
+  useRef,
+} from '#libs/hooks/hooks.js';
+import { type JournalEntryGetAllItemResponseDto } from '#packages/journal/journal.js';
+import { type UserAuthResponseDto } from '#packages/users/users.js';
+import { SAVE_NOTE_TIMEOUT } from '#pages/journal/libs/constants/constants.js';
+import { type NoteContent } from '#pages/journal/libs/types/types.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
+
+import styles from './styles.module.scss';
+
+const Note: React.FC = () => {
+  const { userId, selectedJournalEntry } = useAppSelector(
+    ({ auth, journal }) => {
+      return {
+        userId: (auth.authenticatedUser as UserAuthResponseDto).id,
+        selectedJournalEntry:
+          journal.selectedJournalEntry as JournalEntryGetAllItemResponseDto,
+      };
+    },
+  );
+
+  const { isDirty, control } = useAppForm({
+    defaultValues: {
+      title: selectedJournalEntry.title,
+      text: selectedJournalEntry.text,
+    },
+    mode: 'onChange',
+  });
+
+  const { field: titleField } = useFormController({ name: 'title', control });
+  const { field: textField } = useFormController({ name: 'text', control });
```

Comment:
Let's split note component, this one is responsible for many targets

Reply:
I am not sure that I understand what you mean. Could you explain this,please?

Reply:
Sorry, I don't understand how to split this logic. As for me, it looks pretty solid.

Reply:
let's leave it as it is